.github/workflows/debian-builder-workflow.yaml

```diff
@@ -288,8 +296,23 @@ jobs:
         working-directory: ./build/machinekit-hal  
 
       - name: Sign the package with Machinekit Builder Signer Key
-        if: github.event_name == 'push'
-        run: echo "Not implemented yet"
+        if: github.event_name == 'push' && needs.prepareState.outputs.HasSigningKey == 'true'
+        run: |
+          mkdir ${ROOT_DIR}/secrets_mountpoint
+          sudo mount -t tmpfs -o size=50m tmpfs ${ROOT_DIR}/secrets_mountpoint
+          echo "$SIGNING_KEY" > ${ROOT_DIR}/secrets_mountpoint/key.gpg
+          echo "
+          #!/bin/bash -e
+          gpg --homedir /secrets_mountpoint/.gnupg -v --batch --import '/secrets_mountpoint/key.gpg'
+          find /machinekit-hal -maxdepth 1 -name *.deb -print0 | xargs -0 -n1 -t -I '{}' dpkg-sig --gpg-options '--homedir /secrets_mountpoint/.gnupg' --sign builder '{}'
+          " > ${ROOT_DIR}/secrets_mountpoint/doSignPackages.sh
+          docker run --rm -v "${ROOT_DIR}/secrets_mountpoint:/secrets_mountpoint" -v "$(pwd):/machinekit-hal" ${DOCKER_IMAGE} /bin/bash /secrets_mountpoint/doSignPackages.sh
+          sudo umount ${ROOT_DIR}/secrets_mountpoint
+        env:
+          ROOT_DIR: ${{ github.workspace }}
+          SIGNING_KEY: ${{ secrets.SIGNING_KEY }}
+          DOCKER_IMAGE: 'docker.pkg.github.com/${{ github.repository }}/${{ env.ImageNameRoot }}${{ matrix.architecture }}_${{ matrix.osVersionNumber }}:latest'
```

Comment:
I don't know if the project have a standard coding style stated anywhere, but other projects I work with recommend line length limits.  Up to you whether to reformat or not.

Reply:
No problem, @zultron, I will repair it in next iteration. I also have no idea if some standards for line length exist and how up to date they are with today monitor sizes.

The bash should not be a problem. Not sure about YAML. Github's parser is not that standard conforming.